dessia_common/excel_reader.py

```diff
@@ -0,0 +1,503 @@
+        dict_keys = []
+        if "Dict" in cell.value:
+            actual_value = {}
+            sheet = self.workbook[target]
+            column = sheet["A"]
+            for row in column[3:]:
+                dict_keys.append(row.value.split('.')[1])
+
+            for key, value in zip(dict_keys, values):
+                actual_value[key] = value
+
+            return actual_value
+
+        if isinstance(values, list) and (not any(type_ in cell.value for type_ in container)):
+            actual_value = values[0]
+        else:
+            actual_value = values
+
+        return actual_value
+
+    @staticmethod
+    def get_attributes_and_types(obj_class):
+        """
+        Retrieve attributes and their corresponding types from the __init__ method of a class.
+
+        Args:
+        - obj_class: The class object.
+
+        Returns:
+        - dict: A dictionary containing attributes and their respective types defined in the class constructor.
+        """
+        init_signature = inspect.signature(obj_class.__init__)
+        init_parameters = init_signature.parameters
+
+        attribute_info = {}
+
+        for attr_name, param in init_parameters.items():
+            if attr_name != 'self':
+                attr_info = {
+                    'type': param.annotation if param.annotation != param.empty else None,
+                    'default_value': param.default if param.default != param.empty else "empty"
+                }
+                attribute_info[attr_name] = attr_info
+
+        return attribute_info
+
+    @staticmethod
+    def get_data(value, attributes, init_attributes):
+        """
+        Extracts relevant data based on attributes and initial attributes for object instantiation.
+
+        Args:
+        - value (list): List containing values corresponding to attributes.
+        - attributes (list): List of attributes related to the object.
+        - init_attributes (list): List of initial attributes for object instantiation.
+
+        Returns:
+        - dict: Dictionary containing extracted object data based on matching attributes and initial attributes.
+        """
+        detect_missing_attribute, attribute = missed_attribute(attributes, init_attributes)
+        if not detect_missing_attribute:
+            raise ValueError(f"Missing attribute '{attribute}' doesn't have a default value.")
+
+        object_data = {}
+        for i, attribute in enumerate(attributes):
+            if attribute in init_attributes:
+                if isinstance(value[i], str):
+                    try:
+                        object_data[attribute] = literal_eval(value[i])
+                    except (ValueError, SyntaxError):
+                        object_data[attribute] = value[i]
+                else:
+                    object_data[attribute] = value[i]
+        return object_data
+
+    def create_object(self, obj_class, value_set, attributes, initial_attributes):
```

Comment:
This could take a ExcelDataExtract as an argument OR be a method of it. See further comments

Reply:
As I need value_set (updated data) I keep this method in ExcelReader class, but I made the requested changes before